sandbox/pc2syscheck.sh

```diff
@@ -0,0 +1,94 @@
+#!/bin/bash
+# Copyright (C) 1989-2023 PC2 Development Team: John Clevenger, Douglas Lane, Samir Ashoo, and Troy Boudreau.
+#  
+# File:    pc2syscheck.sh
+# Purpose: Check host system to make sure it's setup properly for sandbox usage
+# 
+# Author: John Buck
+
+# FAIL_RETCODE_BASE is 128 + 64 + xx
+# 128 = system error, like signal
+# 64 = biggest used signal
+FAIL_RETCODE_BASE=192
+FAIL_EXIT_CODE=$((FAIL_RETCODE_BASE+43))
+FAIL_NO_ARGS_EXIT_CODE=$((FAIL_RETCODE_BASE+44))
+FAIL_INSUFFICIENT_ARGS_EXIT_CODE=$((FAIL_RETCODE_BASE+45))
+FAIL_INVALID_CGROUP_INSTALLATION=$((FAIL_RETCODE_BASE+46))
+FAIL_MISSING_CGROUP_CONTROLLERS_FILE=$((FAIL_RETCODE_BASE+47))
+FAIL_MISSING_CGROUP_SUBTREE_CONTROL_FILE=$((FAIL_RETCODE_BASE+48))
+FAIL_CPU_CONTROLLER_NOT_ENABLED=$((FAIL_RETCODE_BASE+49))
+FAIL_MEMORY_CONTROLLER_NOT_ENABLED=$((FAIL_RETCODE_BASE+50))
+FAIL_MEMORY_LIMIT_EXCEEDED=$((FAIL_RETCODE_BASE+51))
+FAIL_TIME_LIMIT_EXCEEDED=$((FAIL_RETCODE_BASE+52))
+FAIL_WALL_TIME_LIMIT_EXCEEDED=$((FAIL_RETCODE_BASE+53))
```

Comment:
Boy, I sure wish the above constants could be defined in ONE place rather than being spread across multiple (at least three) different files.  This just asks for issues down the road when somebody changes ONE of them...

Reply:
Yeah.  One for the java code, one (or two) for the linux scripts, and one for the bat/cmd file.  Ideas?

Reply:
I think this is a "someday".